AsyncRx/src/AsyncObserver.fs

```diff
@@ -86,3 +86,18 @@ module AsyncObserver =
             }
         }
 
+    let autoDetachObserver (obv: IAsyncObserver<'TSource>) : IAsyncObserver<'TSource>*(Async<IAsyncRxDisposable>->Async<IAsyncRxDisposable>) =
+        let disposables : ResizeArray<IAsyncRxDisposable> = ResizeArray ()
+        let cancel () = async {
+            for disp in disposables do
+                do! disp.DisposeAsync ()
+        }
+        let safeObv = AsyncDisposable.Create cancel |> safeObserver obv
+
+        // Auto-detaches (disposes) the disposable when the observer completes with success or error.
+        let autoDetach (disposable: Async<IAsyncRxDisposable>) = async {
+            let! disp = disposable
+            disposables.Add disp
+            return disp
```

Comment:
Not sure, but shouldn't it return `AsyncDisposable.Create cancel`?

Reply:
The auto-detach observer may dispose multiple disposables when it closes, but those individual disposables should not dispose each other. So it's just a pass-through, `id` function with a side-effect. Then you can do `obv |> source.SubscribeAsync |> autoDetach`

Reply:
I see, you're right, thanks.